.circleci/config.yml

```diff
@@ -1,74 +1,115 @@
-          # when lock file changes, use increasingly general patterns to restore cache
-          - gradle-repo-v5-{{ .Branch }}-{{ checksum "gradle.properties" }}
-          - gradle-repo-v5-{{ .Branch }}-
-          - gradle-repo-v5-
-    - run:
-        name: Download uncached dependencies
-        command: ./gradlew dependencies
-        no_output_timeout: 1h
-
-    - save_cache:
-        paths:
-          - ~/.gradle
-        key: gradle-repo-v5-{{ .Branch }}-{{ checksum "gradle.properties" }}
-
-    # Compile
-    - run:
-        name: Compile
-        command: xvfb-run ./gradlew -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildAccess=opensource -PbuildTag=$CIRCLE_TAG --refresh-dependencies --info --stacktrace clean assemble
+    environment:
+      _JAVA_OPTIONS: "-Xmx3g"
+      GRADLE_OPTS: "-Dorg.gradle.daemon=false"
 
-    # Test
-    - run:
-        name: Run tests
-        command: xvfb-run ./gradlew -PbuildAccess=opensource -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildTag=$CIRCLE_TAG --info --stacktrace test
+jobs:
+  checkout_run_test:
+    executor: openjdk_executor
+    description: "Perform all setup steps"
+    steps:
+        - checkout
+        - run:
+            name: Create output directories
+            command: "mkdir -vp ~/circleci-artifacts ~/circleci-test-results ~/circleci-test-results/junit/"
+        
+        - restore_cache:
+            keys:
+                - gradle-repo-v5-{{ .Branch }}-{{ checksum "gradle.properties" }}
+                - gradle-repo-v5-{{ .Branch }}-
+                - gradle-repo-v5-
+        - run:
+            name: Download uncached dependencies
+            command: ./gradlew dependencies
+            no_output_timeout: 1h
+        
+        - save_cache:
+            paths:
+                - ~/.gradle
+            key: gradle-repo-v5-{{ .Branch }}-{{ checksum "gradle.properties" }}
+        
+        - run:
+            name: Compile
+            command: xvfb-run ./gradlew -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildAccess=opensource -PbuildTag=$CIRCLE_TAG --refresh-dependencies --info --stacktrace clean assemble
+        
+        # Test
+        - run:
+            name: Run tests
+            command: xvfb-run ./gradlew -PbuildAccess=opensource -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildTag=$CIRCLE_TAG --info --stacktrace test
+        
+        - run:
+            name: Collect test results
+            command: find . -type f -regex ".*/build/test-results/.*xml" -exec cp -v {} ~/circleci-test-results/junit/ \;
+        - run:
+            name: Collect aritfacts
+            command: '[ ! -d build/reports ] || cp -R build/reports ~/circleci-artifacts/'
+        
+          # Teardown
+          # Save test results
+        - store_test_results:
+            path: ~/circleci-test-results
+          # Save artifacts
+        - store_artifacts:
+            path: ~/circleci-artifacts
+        - store_artifacts:
+            path: ~/circleci-test-results
+        
+        - run: rm -rf .git
```

Comment:
Why was removing the `.git` directory necessary? Just curious.

Reply:
Just to reduce the size of the workspace before running the persist to workspace command.

Reply:
Doesn't the `paths` property already exclude other directories from being persisted, including `.git`?

Reply:
`.git` is inside `project` folder, so it isn't excluded by default.